### ISSUE TITLE
Title: Why is the guidance loss gradient negated twice in DDIMSamplerWithGrad?

### ISSUE THREAD

**Issue by AssafSinger94** [dangeng/motion_guidance#7]
Hey Daniel, great work!
In `ddim_with_grad.DDIMSamplerWithGrad.sample()`, the gradient of the guidance energy appears to be negated twice, as shown in the code snippet below:
```
# Get gradient of energy function
energy, info_loss = guidance_energy(recons_image, src_img)
grad = torch.autograd.grad(energy, noisy_latent_grad)[0]
grad = -grad * guidance_weight * guidance_schedule[i]

...

# Update noise estimate with guidance gradiaent
    # TODO: This may be a bug, try not multiplying here. But need new hyperparams probs
e_t = e_t - sqrt_one_minus_at * grad.detach()   
```

This effectively applies the negative sign twice—once during the gradient computation, and again during the update step. Shouldn’t the gradient be negated only once?
Wouldn’t the correct formulation be simply:  `e_t = e_t - sqrt_one_minus_at * grad.detach()` without the earlier negation? Or is there a specific reason for this double negative?

**Comment by dangeng** [comment]
Basically here I'm using the method from [this paper](https://arxiv.org/pdf/2302.07121) (section 3.1). The `grad` term should point in the direction that encourages the behavior you want (in this case matching the given flow map), and so should minimize the energy/loss, so we have a negative in the first line. The second negation just comes from the formula in the paper (intuitively, this is another double negation... we'll be removing the predicted noise `e_t` from our latents, so we need to _subtract_ from `e_t` the direction we want to move towards, so that the latents move in the desired direction). I agree I could have just removed the negations, but I guess I wanted to keep consistent with the notation from the paper. Let me know if you've got other questions!
